Empire/EmpireManager.cpp

```diff
@@ -80,10 +87,18 @@ std::string EmpireManager::Dump() const {
         switch (entry.second) {
         case DIPLO_WAR:     retval += "War";    break;
         case DIPLO_PEACE:   retval += "Peace";  break;
+        case DIPLO_ALLIED:  retval += "Allied";  break;
         default:            retval += "?";      break;
         }
         retval += "\n";
     }
+    retval += "Diplomatic Messages:\n";
+    for (auto message : m_diplomatic_messages) {
```

Comment:
This creates copies of whatever an entry in m_diplomatic_messages is.  Better use `auto& var` or `const auto& var` when `var` is not modified in the loop.